src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -638,13 +639,22 @@ public void run() {
                     LOG.info("My election bind port: " + addr.toString());
                     setName(addr.toString());
                     ss.bind(addr);
+                    ss.setSoTimeout(10 * 1000); // Ten seconds
```

Comment:
i also think the timeout not need to be set. i just follow [this](https://github.com/apache/karaf/pull/50/commits/0349d582c4899f19ad73ee37c8c688660cbc7354)  simply.